src/Stryker.Core/Stryker.Core/Mutants/MutationContext.cs

```diff
@@ -164,6 +170,11 @@ public StatementSyntax InjectBlockLevel(StatementSyntax mutatedNode, StatementSy
             var wrapper = needReturn
                 ? (Func<ExpressionSyntax, StatementSyntax>)SyntaxFactory.ReturnStatement
                 : SyntaxFactory.ExpressionStatement;
+            // TODO: check is this has any actual benefit
+            if (_store.HasStatementLevel)
+            {
+                mutatedNode = _store.PlaceStatementMutations(mutatedNode, m => wrapper(originalNode.InjectMutation(m)));
```

Comment:
Is this not confusing that `InjectBlockLevelExpressionMutation` will also execute `PlaceStatementMutations`? I think now this method has 2 resposibilities and should be named like `InjectExpressionMutations`.

Reply:
Confusion is in the eye of the beholder, so if you feel this is confusing, it is.
That being said, the purpose of this method is to inject all mutations that should be part of a syntax block, this include any statement level mutation. The current version assumed that there was no pending statement level mutation, since a block contains at least 1 statement, or cannot be mutated. But, this is not true when dealing with block generated with expression body conversion.

But, I need to remove the `TODO`: yes, it has benefits 😄